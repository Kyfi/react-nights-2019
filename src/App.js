import React from 'react'
import { Link } from 'react-router-dom'
import Urls from './constants/urls'

const App = () => (
  <div>
    <nav>
      <Link to={Urls.index}>Homepage</Link>
      <span> |</span>
      <Link to={Urls.signUp}>Sign Up</Link>
      <span> |</span>
      <Link to={Urls.signIn}>Sign In</Link>
      <span> |</span>
      <Link to={Urls.cart}>Cart</Link>
    </nav>
    <h1>E-Commerce app</h1>
  </div>
)

export default App
