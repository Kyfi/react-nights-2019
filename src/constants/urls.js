const url = {
  index: '/',
  signUp: '/sign_up',
  signIn: '/sign_in',
  cart: '/cart',
  productDetail: productId => `/${productId}`,
}

export default url
