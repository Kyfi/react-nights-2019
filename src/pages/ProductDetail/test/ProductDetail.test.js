import React from 'react'
import { waitForElement } from 'react-testing-library'
import 'jest-styled-components'

import { App } from '../../../App'
import urls from '../../../constants/urls'
import { renderWithRouter } from '../../../utilsTest/render'
import { mockFetchProduct } from '../../../utilsTest/mockHelpers'

describe('[pages] ProductDetail', () => {
  const productId = 1

  describe('when loading', () => {
    it('should render correctly', () => {
      const renderer = renderWithRouter(<App />, urls.productDetail(productId))
      expect(renderer.container).toMatchSnapshot()
    })
  })

  describe('when product loaded', () => {
    mockFetchProduct()

    it('should render correctly', async () => {
      const renderer = renderWithRouter(<App />, urls.productDetail(productId))
      await waitForElement(() => renderer.getByTestId('product-detail'))
      expect(renderer.container).toMatchSnapshot()
    })
  })
})
