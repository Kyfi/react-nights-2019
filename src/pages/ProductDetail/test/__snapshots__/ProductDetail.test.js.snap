// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[pages] ProductDetail when loading should render correctly 1`] = `
.c5 {
  padding: 2rem;
}

.c0 {
  position: relative;
  z-index: 1;
}

.c1 {
  background: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  box-shadow: 0 0 5rem -1.3rem rgb(150,150,150);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #252522;
  background-color: #fff;
  padding: 20px;
  border: 0;
  border-right: 1px solid lightgrey;
  white-space: nowrap;
  font-weight: 300;
}

.c3:hover {
  color: #6364d9;
}

.c6 {
  width: 100%;
  padding: 5rem 0;
  padding: 5rem 0;
}

.c7 {
  width: 5rem;
  height: 5rem;
  margin: 0 auto;
}

.c8 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  height: 100%;
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  width: 100%;
}

.c9 {
  stroke: #ef0d33;
  stroke-dasharray: 1,200;
  stroke-dashoffset: 0;
  -webkit-animation: iwIXhf 1.5s ease-in-out infinite;
  animation: iwIXhf 1.5s ease-in-out infinite;
  stroke-linecap: round;
}

<div>
  <div
    class="Toastify"
  />
  <div
    class="c0"
  >
    <nav
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          Product List
        </a>
      </div>
      <div
        class="c4"
      >
        <a
          class="c3"
          href="/sign_in"
        >
          Sign In
        </a>
        <a
          class="c3"
          href="/sign_up"
        >
          Sign Up
        </a>
        <a
          class="c3"
          href="/cart"
        >
          My Cart
        </a>
      </div>
    </nav>
  </div>
  <div
    class="c5"
  >
    <div
      class="c6"
      data-testid="loader"
    >
      <div
        class="c7"
      >
        <svg
          class="c8"
          viewBox="25 25 50 50"
        >
          <circle
            class="c9"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            stroke-miterlimit="10"
            stroke-width="4"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`[pages] ProductDetail when product loaded should render correctly 1`] = `
.c5 {
  padding: 2rem;
}

.c0 {
  position: relative;
  z-index: 1;
}

.c1 {
  background: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  box-shadow: 0 0 5rem -1.3rem rgb(150,150,150);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #252522;
  background-color: #fff;
  padding: 20px;
  border: 0;
  border-right: 1px solid lightgrey;
  white-space: nowrap;
  font-weight: 300;
}

.c3:hover {
  color: #6364d9;
}

.c11 {
  background: #ef0d33;
  padding: 1rem;
  margin-top: 0.5rem;
  border: 1px solid #ef0d33;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c11:focus {
  outline: none;
}

.c11:hover {
  background: transparent;
  color: #ef0d33;
}

.c9 {
  color: #ef0d33;
  font-size: 1.8rem;
  margin-top: 2rem;
}

.c8 {
  max-width: 100%;
}

.c6 {
  overflow: hidden;
}

.c7 {
  font-size: 2.6rem;
  text-transform: uppercase;
  font-weight: bold;
}

.c10 {
  color: #252522;
  font-size: 1.4rem;
  line-height: 1.56;
  margin-top: 2rem;
}

@media (min-width:600px) {
  .c8 {
    max-height: 20rem;
  }
}

@media (min-width:900px) {
  .c6 {
    height: 6.7rem;
  }
}

<div>
  <div
    class="Toastify"
  />
  <div
    class="c0"
  >
    <nav
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          Product List
        </a>
      </div>
      <div
        class="c4"
      >
        <a
          class="c3"
          href="/sign_in"
        >
          Sign In
        </a>
        <a
          class="c3"
          href="/sign_up"
        >
          Sign Up
        </a>
        <a
          class="c3"
          href="/cart"
        >
          My Cart
        </a>
      </div>
    </nav>
  </div>
  <div
    class="c5"
  >
    <div
      data-testid="product-detail"
    >
      <div
        class=""
      >
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            First Test Product
          </h3>
        </div>
        <div
          class=""
        >
          <img
            alt="First Test Product image"
            class="c8"
            src="https://img.commercelayer.io/skus/1.png"
          />
        </div>
        <div
          class="c9"
        >
          $10
        </div>
        <div
          class="c10"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam pellentesque in neque vitae tincidunt. In gravida eu ipsum non condimentum. Curabitur libero leo, gravida a dictum vestibulum, sollicitudin vel quam.
        </div>
        <button
          class="c11"
          type="button"
        >
          Add to Cart
        </button>
        <hr />
        <a
          href="/"
        >
          Back to Product List
        </a>
      </div>
    </div>
  </div>
</div>
`;
